doc/Architecture/arc-tec_x-road_technologies.md

```diff
@@ -0,0 +1,137 @@
+
+ **Technology**   | **Web Server**   | **Configuration Processor**   | **Signer**   | **Configuration Client**
+----------------- | ---------------- | ----------------------------- | ------------ | ------------------------
+ Java 8           |                  | X                             | X            | X
+ Logback          |                  | X                             | X            | X
+ Akka 2.X         |                  | X                             | X            |
+ nginx            | X                |                               |              |
+ upstart          | X                | X                             | X            | X
+ PKCS \#11\[[2](#Ref_2)\]   |                |                             | X          |
+
+<a id="Ref_2" class="anchor"></a>
+\[2\] The use of hardware cryptographic devices requires that a PKCS \#11 driver is installed and configured in the system.
+
+
+## 4 Security server technologies
+
+[Table 4](#Ref_Technology_matrix_of_the_security_server) presents the list of technologies used in the security server and the mapping between technologies and security server components.
+
+<a id="Ref_Technology_matrix_of_the_security_server" class="anchor"></a>
+Table 4. Technology matrix of the security server
+
+ **Technology**     | **Signer**   | **Proxy**   | **Password Store**   | **Message Log**   | **Metadata Services**   | **Database**   | **Configuration Client**   | **User Interface**   | **Servlet Engine**   | **Monitor**   | **Environmental Monitoring Service**   | **Operational Monitoring Buffer**   | **Operational Monitoring Services**
+------------------- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | ---
+ Java 8             | X   | X   |     | X   | X   |     | X   | X   | X   | X   | X   | X   | X
+ C                  |     |     | X   |     |     |     |     |     |     |     |     |     |
+ Logback            | X   | X   |     | X   | X   |     | X   | X   |     |     | X   | X   | X
+ Akka 2.X           | X   | X   |     | X   |     |     |     | X   |     | X   | X   | X   |
+ Jetty 9            |     |     |     |     |     |     |     |     | X   |     |     |     |
+ JRuby 1.7          |     |     |     |     |     |     |     | X   |     |     |     |     |
+ Javascript         |     |     |     |     |     |     |     | X   |     |     |     |     |
+ PostgreSQL 9.3     |     |     |     |     |     | X   |     |     |     |     |     |     |
+ PAM                |     |     |     |     |     |     |     |     | X   |     |     |     |
+ Liquibase          |     |     |     |     |     | X   |     |     |     |     |     |     |
+ upstart            | X   | X   |     |     |     |     | X   |     | X   |     |     |     |
+ PKCS \#11\[[2](#Ref_2)\]       | X   |     |     |     |     |     |     |     |     |     |     |     |
+ Dropwizard Metrics |     |     |     |     |     |     |     |     |     | X   |     |     |
+
```

Comment:
Shouldn't OP monitoring daemon tech matrix be transferred as well? A bit odd to leave it out when It is mentioned in the totals. The document is under a different dir but still an architecture doc.